### PR TITLE
Port to python 3

### PR DESCRIPTION
**Pull request by ssjunnebo**
* Compatible with both python 2 and 3
* Tests set to run in both versions
* Tested locally, so a proper validation would be good

### CODE REVIEW

**Review by aanil — APPROVED** [review]
Looks ok to me.

**Review by matrulda — COMMENTED** [review]
Looks good! 👍 Left a few comments that you can have a look at.

**Comment by matrulda** [comment]
Oh another thing. Did the code climate check fail before? 🤔

File: ngi_pipeline/conductor/classes.py [hunk 1]
```diff
@@ -62,11 +63,9 @@ def __eq__(self, other):
             self._subitem_type == other._subitem_type,
             len(self._subitems) == len(other._subitems),
             all(
-                map(
-                    lambda (s, o): s == o,
-                    zip(
+                [s_o[0] == s_o[1] for s_o in zip(
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/conductor/classes.py:66]: According to the [porting guide](https://portingguide.readthedocs.io/en/latest/iterators.html?highlight=map#new-behavior-of-map-and-filter) list comprehensions should be faster?

File: ngi_pipeline/conductor/flowcell.py [hunk 2]
```diff
@@ -290,7 +291,6 @@ def setup_analysis_directory_structure(fc_dir, projects_to_analyze,
                                                      sample_name,
                                                      fc_full_id,
                                                      fq_file,
-                                                     libprep_name,
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/conductor/flowcell.py:293]: True! 👍

File: ngi_pipeline/engines/piper_ngi/launchers.py [hunk 3]
```diff
@@ -251,7 +251,7 @@ def collect_files_for_sample_analysis(project_obj, sample_obj,
                           project_obj.project_id, project_obj.base_path)
     sample_obj = proj_obj.add_sample(sample_obj.name, sample_obj.dirname)
     # filter out index files from the analysis
-    for fastq_path in filter(lambda f: not is_index_file(f), fastq_files_on_filesystem):
+    for fastq_path in [f for f in fastq_files_on_filesystem if not is_index_file(f)]:
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/engines/piper_ngi/launchers.py:254]: Same reason as for [map()](https://portingguide.readthedocs.io/en/latest/iterators.html?highlight=map#higher-order-functions-vs-list-comprehensions), though it probably doesn't matter much in this case.

**Comment by ssjunnebo** [comment]
> Looks good! 👍 Left a few comments that you can have a look at.

Awesome, thank you! I've changed the libprep_name. Let me know if you think we should keep map() and filter() instead of the list comprehensions 🙂

File: ngi_pipeline/conductor/classes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (matrulda) [reply ngi_pipeline/conductor/classes.py:66]: Yeah you are right. I got it backwards.

**Review by matrulda — APPROVED** [review]
Great! I think we should keep list comprehensions 👍  

Ah ok. Weird that they start acting up now 🤷‍♀️ 
I'm ready to merge if you are.

**Comment by ssjunnebo** [comment]
Great! Sure, I think we can go ahead and merge. It will need some through testing on irma, so it's good timing that we have a longer deployment this time 😄

**Pull request merged**